pkg/plugins/envvars/envvars.go

```diff
@@ -110,6 +112,15 @@ func (s *Service) GetConfigMap(ctx context.Context, pluginID string, _ *auth.Ext
 		m[backend.ConcurrentQueryCount] = strconv.Itoa(s.cfg.ConcurrentQueryCount)
 	}
 
+	if s.cfg.UserFacingDefaultError != "" {
+		m[backend.UserFacingDefaultError] = s.cfg.UserFacingDefaultError
+	}
+
+	m[backend.SQLRowLimit] = strconv.FormatInt(s.cfg.DataProxyRowLimit, 10)
+	m[backend.SQLMaxOpenConnsDefault] = strconv.Itoa(s.cfg.SqlDatasourceMaxOpenConnsDefault)
+	m[backend.SQLMaxIdleConnsDefault] = strconv.Itoa(s.cfg.SqlDatasourceMaxIdleConnsDefault)
+	m[backend.SQLMaxConnLifetimeSecondsDefault] = strconv.Itoa(s.cfg.SqlDatasourceMaxConnLifetimeDefault)
```

Comment:
do any of these make sense to set if they are 0? For example, a row limit of 0 sounds like an error

Reply:
@andresmgot 

the sql-max-ones all have a meaningful use of a `0` value.  (max-open and max-lifetime use `0` to say it is unlimited, and `max-idle=0` means that do not keep any idle connections open).

for rowLimit, it's eventually used in https://github.com/grafana/grafana-plugin-sdk-go/blob/6c53dc94dcc80e338c46e2f96a05fff73fcdb003/data/sqlutil/sql.go#L22 , it says: "If rowLimit is less than 0, there is no limit." . so `rowlimit=0` will still make a request, and create a response, but it will contain `0` rows and a warning message. so, we could say setting this to `0` has no sense. (it's funny that a negative-value has sense 😄 )

my problem is that ... there is no possible situation where that thing will be zero, right? :
1. users will not set it to zero, because it makes no sense
2. because of https://github.com/grafana/grafana/blob/8586893731aa200a5b10c07621988686abe10729/pkg/setting/setting_data_proxy.go#L24 , if it is unset or some not-valid-number, it will be set to 1million

so, if i add an `if rowLimit != 0`,  that `if` will only do something in unit-tests, not real code... 
but maybe i'm looking at this wrong, WDYT?

Reply:
👍 , i changed it so that  `rowlimit` is only set when it's non-zero.